### PR TITLE
Feature/log beer functionality#6

### LINKED ISSUE #6: Log Beers functionality implementation (Log registry)

### PR DESCRIPTION
**Pull request by dannharo**
closes #6 
Adding the functionality of log your beers

Add controller, model and views to the log of beers
Add home view

### CODE REVIEW

File: app/controllers/beer_logs_controller.rb [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+      flash[:alert] = @beerlog.errors.full_messages.join
+      render :edit
+    end
+  end
+  def destroy
+    @beerlog.destroy
+    if @beerlog.destroyed?
+      redirect_to user_root_path
+    else
+      render :edit
```
Comment (jclopezdev) [inline app/controllers/beer_logs_controller.rb]: You could add a flash message when the deletion is not successful

File: app/controllers/beer_logs_controller.rb [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+      redirect_to user_root_path
+    else
+      flash[:alert] = @beerlog.errors.full_messages.join
+      render :edit
+    end
+  end
+  def destroy
+    @beerlog.destroy
+    if @beerlog.destroyed?
+      redirect_to user_root_path
```
Comment (jclopezdev) [inline app/controllers/beer_logs_controller.rb]: You could add a flash message when the deletion is successful

File: app/controllers/beer_logs_controller.rb [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  end
+
+  def update
+    if @beerlog.update_attributes(beer_log_params)
+      redirect_to user_root_path
+    else
+      flash[:alert] = @beerlog.errors.full_messages.join
+      render :edit
+    end
+  end
```
Comment (vicmaster) [inline app/controllers/beer_logs_controller.rb]: Please add a blank line between methods

File: app/controllers/beer_logs_controller.rb [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    else
+      render :edit
+    end
+  end
+
+  private
+  def beer_log_params
+    params.require(:beer_log).permit(:beer_id, :quantity)
+  end
+  def set_beer_log
```
Comment (vicmaster) [inline app/controllers/beer_logs_controller.rb]: Please add a blank line between methods

File: app/controllers/beer_logs_controller.rb [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+  def create
+    @beerlog = current_user.beer_logs.new(beer_log_params)
+    if @beerlog.save
+      flash[:success] =  "Created!"
+     redirect_to user_root_path
+    else
+      flash[:alert] = @beerlog.errors.full_messages.join
+      render :new
+    end
+
```
Comment (vicmaster) [inline app/controllers/beer_logs_controller.rb]: Remove this extra blank space

File: app/controllers/home_controller.rb [hunk 6]
```diff
@@ -1,9 +1,13 @@
...
   before_action :authenticate_user!
+  before_action :set_data
 
   def index
+    @beerlog = BeerLog.new
+  end
 
+  private
+  def set_data
+    @beerlogs = BeerLog.where("user_id = ? and created_at >= ?", current_user.id,Time.zone.now.beginning_of_day).order(:id)
```
Comment (jclopezdev) [inline app/controllers/home_controller.rb]: Remember the 80 character limit per line.

```
@beerlogs = BeerLog.where("user_id = ? and created_at >= ?", 
  current_user.id,Time.zone.now.beginning_of_day).order(:id)
```

File: app/controllers/beer_logs_controller.rb [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+  def destroy
+    @beerlog.destroy
+    if @beerlog.destroyed?
+      redirect_to user_root_path
+    else
+      render :edit
+    end
+  end
+
+  private
```
Comment (vicmaster) [inline app/controllers/beer_logs_controller.rb]: Please add a blank line between `private` and the private methods

File: app/controllers/home_controller.rb [hunk 8]
```diff
@@ -1,9 +1,13 @@
 class HomeController < ApplicationController
   before_action :authenticate_user!
+  before_action :set_data
 
   def index
+    @beerlog = BeerLog.new
+  end
 
+  private
```
Comment (vicmaster) [inline app/controllers/home_controller.rb]: Please add a blank line between `private` and the private methods

File: app/helpers/beer_logs_helper.rb [hunk 9]
```diff
@@ -0,0 +1,2 @@
+module BeerLogsHelper
```
Comment (vicmaster) [inline app/helpers/beer_logs_helper.rb]: Do this helper is being used?

File: app/views/beer_logs/edit.html.haml [hunk 10]
```diff
@@ -0,0 +1,20 @@
...
+          %label.active Beer
+          = form.select(:beer_id, options_for_select(Beer.all.map{|s|[s.name,s.id]}, @beerlog.beer_id), id: "select-beast")
+      .row
+        .input-field.col.s4
+          %label.active{for: "quantity"} Quantity of Beers
+          = form.number_field :quantity, value: @beerlog.quantity.to_i
+        .input-field.col.s4.right-align
+          = form.submit  class: 'btn waves-effect blue lighten-1'
+        .input-field.col.s4.right-align
+          = link_to "Delete record",beer_log_path(@beerlog), method: :delete, class: 'btn waves-effect red lighten-1', data: { confirm: 'Are you sure?' }
```
Comment (jclopezdev) [inline app/views/beer_logs/edit.html.haml]: Remember the 80 character limit per line.

File: app/views/beer_logs/new.html.haml [hunk 11]
```diff
@@ -0,0 +1,15 @@
+%h2
+  Add the beer you've drunk!!
+.row
+  .col.s12
+    = form_with model: @beerlog, local: true do |form|
+      .row
+        .input-field.col.s12
+          = form.select(:beer_id, options_for_select(Beer.all.map{|s|[s.name,s.id]}),{include_blank: true}, id: "select-beast")
+          %label.active{for: "beer_id"} Beer
```
Comment (vicmaster) [inline app/views/beer_logs/new.html.haml]: Add spaces between curly braces `{}` like `{ for: 'beer_id' }`

File: spec/helpers/beer_logs_helper_spec.rb [hunk 12]
```diff
@@ -0,0 +1,15 @@
+require 'rails_helper'
```
Comment (vicmaster) [inline spec/helpers/beer_logs_helper_spec.rb]: Remove this test if it's doing anything

File: spec/models/beer_log_spec.rb [hunk 13]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
```
Comment (vicmaster) [inline spec/models/beer_log_spec.rb]: Remove this test if it's doing anything

File: spec/views/beer_logs/edit.html.haml_spec.rb [hunk 14]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
```
Comment (vicmaster) [inline spec/views/beer_logs/edit.html.haml_spec.rb]: Remove this test if it's doing anything

File: spec/views/beer_logs/index.html.haml_spec.rb [hunk 15]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
+
+RSpec.describe "beer_logs/index.html.haml", type: :view do
```
Comment (vicmaster) [inline spec/views/beer_logs/index.html.haml_spec.rb]: Remove this test if it's doing anything

File: spec/views/beer_logs/new.html.haml_spec.rb [hunk 16]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
+
+RSpec.describe "beer_logs/new.html.haml", type: :view do
```
Comment (vicmaster) [inline spec/views/beer_logs/new.html.haml_spec.rb]: Remove this test if it's doing anything

File: spec/views/beer_logs/show.html.haml_spec.rb [hunk 17]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
+
```
Comment (vicmaster) [inline spec/views/beer_logs/show.html.haml_spec.rb]: Add some examples to (or delete)

File: app/models/user.rb [hunk 18]
```diff
@@ -5,9 +5,11 @@ class User < ApplicationRecord
   devise :database_authenticatable, :registerable,
          :recoverable, :rememberable, :trackable,
          :omniauthable, omniauth_providers: [:facebook, :google_oauth2]
+         has_many :beer_logs
   has_attached_file :photo,
     styles: { thumbnail: "300x300>", thumb: "150x150>", small: "20x20" }
   validates_attachment_content_type :photo, content_type: /\Aimage\/.*\z/
+  has_many :beer_logs
```
Comment (edman193) [inline app/models/user.rb]: delete repeated line => has_many :bear_logs

**Pull request merged**